src/bullet.rs

```diff
@@ -0,0 +1,61 @@
+use bevy::prelude::*;
+
+pub struct BulletPlugin;
+
+pub use crate::enemy::*;
+
+impl Plugin for BulletPlugin {
+  fn build(&self, app: &mut App) {
+    app.register_type::<Bullet>()
+       .add_system(move_bullets)
+       .add_system(despawn_bullets)
+       .add_system(bullet_enemy_collision);
+  }
+}
+
+// !!! Debugging
+#[derive(Reflect, Component, Default)]
+#[reflect(Component)]
+pub struct Bullet {
+  pub direction: Vec3,
+  pub speed: f32,
+  pub damage: i32,
+  pub lifetime: Timer // !!! fix?
+}
+
+fn move_bullets(mut bullets: Query<(&Bullet, &mut Transform)>, time: Res<Time>) {
```

Comment:
Maybe it is better to have one system that just moved things and have a component for speed and direction, instead of having 2 moving system one for enemies and one for bullets